test/integration/jobschedulingpreference_test.go

```diff
@@ -0,0 +1,264 @@
+/*
+Copyright 2018 The Federation v2 Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"testing"
+
+	fedv1a1 "github.com/kubernetes-sigs/federation-v2/pkg/apis/core/v1alpha1"
+	fedschedulingv1a1 "github.com/kubernetes-sigs/federation-v2/pkg/apis/scheduling/v1alpha1"
+	clientset "github.com/kubernetes-sigs/federation-v2/pkg/client/clientset/versioned"
+	"github.com/kubernetes-sigs/federation-v2/pkg/schedulingtypes"
+	"github.com/kubernetes-sigs/federation-v2/pkg/schedulingtypes/adapters"
+	"github.com/kubernetes-sigs/federation-v2/test/common"
+	"github.com/kubernetes-sigs/federation-v2/test/integration/framework"
+	batchv1 "k8s.io/api/batch/v1"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	restclient "k8s.io/client-go/rest"
+)
+
+// TestReplicaSchedulingPreference validates basic replica scheduling preference calculations.
+var TestJobSchedulingPreference = func(t *testing.T) {
+	t.Parallel()
+	tl := framework.NewIntegrationLogger(t)
+
+	controllerFixture, fedClient := initJSPTest(tl, FedFixture)
+	defer controllerFixture.TearDown(tl)
+
+	clusters := getClusterNames(fedClient, FedFixture.SystemNamespace)
+	if len(clusters) != 2 {
+		tl.Fatalf("Expected two clusters to be part of Federation Fixture setup")
+	}
+
+	kubeClient := FedFixture.KubeApi.NewClient(tl, "jsp")
+	testNamespace := framework.CreateTestNamespace(tl, kubeClient, "jsp-")
+
+	testCases := map[string]struct {
+		jspSpec  fedschedulingv1a1.JobSchedulingPreferenceSpec
+		expected map[string]adapters.JobSchedulingResult
+	}{
+		"Distribution happens equally in clusters, with no explicit per cluster weights": {
```

Comment:
What is tested in these 3 cases that isn't already tested in the unit tests?  Consider limiting this test to a single golden-path check that the output of the scheduler is correctly translated into the expected placement and override resources.

Reply:
Have reduced the test to a single test validation the integration flow!